src/bundles/helia.js

```diff
@@ -0,0 +1,213 @@
+        console.warn('IPLD Explorer requires access to window.crypto, redirecting to canonical URL that is known to provide it in all browsers')
+        // This error means js-ipfs was loaded in a context that is not marked
+        // as Secure Context by the browser vendor.  (example: *.localhost in
+        // Firefox until https://bugzilla.mozilla.org/show_bug.cgi?id=1220810
+        // is addresssed)
+        // This is difficult to debug for regular user, as Explorer simply fails to load anything from IPFS.
+        // For now, we detect this failure and redirect to canonical version with TLS, so it always works.
+        const url = new URL('https://explore.ipld.io/?x-ipfs-companion-no-redirect')
+        // url.hash = window.location.hash
+        // window.location.replace(url.toString())
+        return
+      }
+      console.log('Failed to initialise helia', error)
+      console.timeEnd('IPFS_INIT')
+      return dispatch({ type: 'IPFS_INIT_FAILED', error })
+    }
+  }
+}
+export default bundle
+
+async function initHelia (ipfsApi) {
+
+  const delegatedClient1 = kuboClient(ipfsApi)
+  const delegatedClient2 = kuboClient({
+    protocol: 'https',
+    port: 443,
+    host: 'node3.delegate.ipfs.io'
+  })
+  const blockstore = new MemoryBlockstore()
+  const datastore = new MemoryDatastore()
+
+  /**
+   * based on https://github.com/ipfs/helia/blob/ed4985677b62021f76541354ad06b70bd53e929a/packages/helia/src/utils/libp2p.browser.ts#L20
+   */
+  const libp2p = await createLibp2p({
+    addresses: {
+      listen: [
+        '/webrtc'
+      ]
+    },
+    peerRouters: [
+      delegatedPeerRouting(delegatedClient1),
+      delegatedPeerRouting(delegatedClient2)
+    ],
+    contentRouters: [
+      ipniContentRouting('https://cid.contact'),
+      delegatedContentRouting(delegatedClient1),
+      delegatedContentRouting(delegatedClient2)
+    ],
+    datastore,
+    transports: [
+      webRTC(),
+      webRTCDirect(),
+      webTransport(),
+      webSockets(),
+      circuitRelayTransport({
+        discoverRelays: 1
+      })
+    ],
+    connectionEncryption: [
+      noise()
+    ],
+    streamMuxers: [
+      yamux(),
+      mplex()
+    ],
+    peerDiscovery: [
+      bootstrap({
+        list: [
+          '/dnsaddr/bootstrap.libp2p.io/p2p/QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN',
+          '/dnsaddr/bootstrap.libp2p.io/p2p/QmQCU2EcMqAqQPR2i9bChDtGNJchTbq5TbXJJ16u19uLTa',
+          '/dnsaddr/bootstrap.libp2p.io/p2p/QmbLHAnMoJPWSCR5Zhtx6BHJX9KiKNN6tpvbUcqanj75Nb',
+          '/dnsaddr/bootstrap.libp2p.io/p2p/QmcZf59bWwK5XFi76CZX8cbJ4BhTzzA3gU1ZjYZcYW3dwt',
+          '/ip4/104.131.131.82/tcp/4001/p2p/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ'
+        ]
+      })
+    ],
+    services: {
+      identify: identifyService(),
+      autoNAT: autoNATService(),
+      pubsub: gossipsub(),
+      dht: kadDHT({
+        clientMode: true
+      }),
+      relay: circuitRelayServer({
+        advertise: true
```

Comment:
:thought_balloon: @SgtPooki  is this right? I thought a browser node can't act as an useful  relay server, especially without https://connectivity.libp2p.io/#webrtc?tab=browser-to-browser

Reply:
@SgtPooki so we have a bug there too?

according to [docs here](https://github.com/libp2p/js-libp2p/blob/ac6e65d1a93ac66b767bee380db61abd3d436c25/doc/CONFIGURATION.md?plain=1#L469-L471) `circuitRelayServer` "makes the node function as a relay server" and I am unsure how would that work in a browser, since you don't have a publicly dialable address (aside from making no sense, since relay should be stable, and helia running on a webpage is ephemeral)

Reply:
I guess so, I'd have to point you at some js-libp2p folks to determine for sure, but yea, we could probably remove this for sure since we're trying to do GETs only

Reply:
removed